client.js

```diff
@@ -149,32 +149,39 @@ class Client {
                         return Promise.reject(new ClientError('Unable to parse response'));
                     }
                 }
-
-                if (response.body.error && response.body.error.message) {
-                    const error = new ApiError(response.body.error.message);
-                    if (response.body.error.code !== 'undefined') {
-                        error.code = response.body.error.code;
-                    }
-                    error.status = response.statusCode;
+                const error = this.parseError(response)
+                if (error) {
                     return Promise.reject(error);
                 }
-
-                if (response.statusCode) {
-                    if (parseInt(response.statusCode/100) === 4) {
-                        const error = new HttpError('Invalid request');
-                        error.status = response.statusCode;
-                        return Promise.reject(error);
-                    }
-                    if (parseInt(response.statusCode/100) === 5) {
-                        const error = new HttpError('Internal error');
-                        error.status = response.statusCode;
-                        return Promise.reject(error);
-                    }
-                }
                 return Promise.resolve(response);
             })
     }
 
+    parseError(response) {
```

Comment:
I think this function can be static

Reply:
But it wouldn't be able to override if it's static ?

Reply:
You can override static functions

https://pastebin.com/iBvWRgGC